src/NuGet.Core/NuGet.PackageManagement/Projects/ProjectJsonNuGetProject.cs

```diff
@@ -220,7 +220,7 @@ public override async Task<(IReadOnlyList<PackageSpec> dgSpecs, IReadOnlyList<IA
 
                 var references = (await ProjectServices
                     .ReferencesReader
-                    .GetProjectReferencesAsync(context.Logger, CancellationToken.None))
+                    .GetProjectReferencesAsync(context?.Logger, CancellationToken.None))
```

Comment:
On line 174, we can see that this code might run when `context` is `null`. On line 248, the code uses `context?.` pattern to guard against `null`. This method call should follow that pattern.

Reply:
I believe passing a `null` for the logger is also problematic. Likely to encounter exceptions later.
Try this out:
```suggestion
                    .GetProjectReferencesAsync(context?.Logger ?? NullLogger.Instance, CancellationToken.None))
```